tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_prplwrt.py

```diff
@@ -27,27 +35,55 @@ def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
         config = kwargs.get("config", kwargs)
+
+
+        self.unique_id = os.getenv("SUDO_USER", os.getenv("USER", ""))
         self.docker_network = config.get("docker_network",
                                          "prplMesh-net-{}".format(self.unique_id))
         self.role = config.get("role", "agent")
-        if self.role == "controller":
-            self.controller_entity = ALEntityPrplWrt(self.name, is_controller=True)
-        else:
-            self.agent_entity = ALEntityPrplWrt(self.name, is_controller=False)
+        self.connection_type = config.get("connection_type", None)
+        self.conn_cmd = config.get("conn_cmd", None)
+
+        self.connection = connection_decider.connection(device=self,
+                                                        conn_type=kwargs['connection_type'],
+                                                        **kwargs)
+        self.connection.connect()
 
+        self.name = "-".join((config.get("name", "rax40"), self.unique_id))
+        self.consoles = [self]
+        self.logfile_read = sys.stdout
+
+        self.wan_network = self.get_docker_subnet(self.docker_network)
```

Comment:
Are you expecting that any boards will be connected with test PC only with WAN port and not with LAN port or both LAN and WAN ports?
I think it's not common case